xbmc/pvr/timers/PVRTimerInfoTag.h

```diff
@@ -636,9 +637,8 @@ class CPVRTimerInfoTag final : public ISerializable
   unsigned int m_iRadioChildTimersRecording = 0;
   unsigned int m_iRadioChildTimersErrors = 0;
 
-  mutable std::shared_ptr<CPVREpgInfoTag> m_epgTag; /*!< epg info tag matching m_iEpgUid. */
+  mutable std::optional<std::shared_ptr<CPVREpgInfoTag>>
```

Comment:
I don't think this is really the correct use of std::optional. Wrapping a smart pointer can already be checked if it contains an object if it nullptr or not. Maybe I'm mistaken but I would think you use std::optional with an object or a smart pointer for delayed initialization, not both.

Reply:
> Wrapping a smart pointer can already be checked if it contains an object if it nullptr or not.

Sure, but not sufficient for my use case. I don't only want to implement lazy initialization, I want lazy eval in combination check only once. Legit result of the eval can be nullptr and exactly for this case I need the std::optional, to know that I already did the check.